controllers/bsl.go

```diff
@@ -252,6 +252,14 @@ func (r *DPAReconciler) updateBSLFromSpec(bsl *velerov1.BackupStorageLocation, d
 				bslSpec.Config["s3Url"] = s3Url
 			}
 		}
+
+		// Since the AWS SDK upgrade in velero-plugin-for-aws, data transfer to BSL bucket fails unless
+		// we specify the checksumAlgorithm. We will set it empty string if checksumAlgorithm is not specified by the user
+		// Setting it to an empty string will default to using the CRC32 algorithm for checksum calculation.
+		if _, exists := bslSpec.Config[checksumAlgorithm]; len(s3Url) > 0 && !exists {
+			bslSpec.Config[checksumAlgorithm] = ""
```

Comment:
I'm thinking we don't necessarily wnat to check s3Url here -- that makes the default situation more confusing to end users. If our goal is to preserve OADP 1.3 functionality, then we should always set this to "" if it doesn't exist in the map.

Reply:
or just to be sure, do an AWS backup in 1.3 and see if it works in 1.4 with checksumAlgorithm? or simulate a disaster scenario?

but I do agree with Scott regarding preserving 1.3 behavior

Reply:
@mateusoliveira43 Good point about checking 1.3 backups on 1.4 -- I suspect it shouldn't matter. using checksum vs. not shouldn't affect the on-disk blobs, as I think this is mainly relevant for transport, but QE does have "test 1.3 backup restored in 1.4" in the test matrix. That should probably be updated to test with and without `checksumAlgorithm` specified.

Reply:
added a comment in JIRA to test this upgrade scenario.